commands/role-request.js

```diff
@@ -0,0 +1,39 @@
+const {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  SlashCommandBuilder,
+} = require("discord.js");
+
+module.exports = {
+  data: new SlashCommandBuilder()
+    .setName("request-role")
+    .setDescription("Request a role."),
```

Comment:
this description isn't technically correct - it *sends* the message that allows choosing a role, not allowing requesting roles.